src/controllers/redeploy.go

```diff
@@ -8,16 +8,15 @@ func redeploy(callbackID string, data map[string]interface{}) {
 	stop(callbackID, data)
 	state := history.GetState(data["git_repo"].(string))
 	if state.Status == "stopped" {
-		if _, ok := data["subdomain"]; !ok {
+		if _, ok := data["subdomain"]; !ok && state.Subdomain != "" {
 			data["subdomain"] = state.Subdomain
 		}
-		if _, ok := data["access"]; !ok {
+		if _, ok := data["access"]; !ok && state.Access != "" {
 			data["access"] = state.Access
 		}
-		if _, ok := data["server_name"]; !ok {
+		if _, ok := data["server_name"]; !ok && state.Server != "" {
 			data["server_name"] = state.Server
 		}
```

Comment:
you need to return if any of these three fields does not get set to a non-empty value otherwise the deploy action would happen anyway

Reply:
@codeboy5 also can you add some logging statements also. you can look at deploy.go or stop.go for reference.

Reply:
I have added separate logging statements for each parameter, I can change it to a single error in the end if you want.
This gives more info to the user